test/e2e/cloud-provider-azure.go

```diff
@@ -0,0 +1,149 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	helmAction "helm.sh/helm/v3/pkg/action"
+	helmLoader "helm.sh/helm/v3/pkg/chart/loader"
+	helmCli "helm.sh/helm/v3/pkg/cli"
+	helmVals "helm.sh/helm/v3/pkg/cli/values"
+	helmGetter "helm.sh/helm/v3/pkg/getter"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// InstallCloudProviderAzureHelmChart installs the official cloud-provider-azure helm chart
+func InstallCloudProviderAzureHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, result *clusterctl.ApplyClusterTemplateAndWaitResult) {
+	By("Waiting for workload cluster kubeconfig secret")
+	Eventually(func() error {
+		client := input.ClusterProxy.GetClient()
+		secret := &corev1.Secret{}
+		key := crclient.ObjectKey{
+			Name:      fmt.Sprintf("%s-kubeconfig", input.ConfigCluster.ClusterName),
+			Namespace: input.ConfigCluster.Namespace,
+		}
+		return client.Get(ctx, key, secret)
+	}, 20*time.Minute, 5*time.Second).Should(Succeed())
+	clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
+	By("Waiting for nodes to come online indicating that the cluster is ready to accept work")
+	Eventually(func() error {
+		clientSet := clusterProxy.GetClientSet()
+		var runningNodes int
+		list, err := clientSet.CoreV1().Nodes().List(ctx, v1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		for _, n := range list.Items {
+			if n.Status.Phase == corev1.NodeRunning {
+				runningNodes++
+			}
+		}
+		if runningNodes > 0 {
+			return nil
+		}
+		return err
+	}, 20*time.Minute, 5*time.Second).Should(Succeed())
+	kubeConfigPath := clusterProxy.GetKubeconfigPath()
+	os.Setenv("KUBECONFIG", kubeConfigPath)
```

Comment:
why do we need to export a var here? Any way we can pass in the kubeconfig to Helm another way so we don't mess with env vars? If not, we should add a defer statement to unset the var.

Reply:
There's gotta be a way to pass in the filepath somewhere in the flow, as the CLI (which uses these libs under the hood) supports that.

```sh
$ helm install --help | grep kubeconfig
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/jackfrancis/.kube/external-cloud-provider-24053.kubeconfig
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/jackfrancis/.kube/external-cloud-provider-24053.kubeconfig
      --kube-context string         name of the kubeconfig context to use
      --kubeconfig string           path to the kubeconfig file
```

I'll figure that out.

Reply:
Replaced `KUBECONFIG` env var w/ object property assignment